### PR TITLE
Fix TableType processing logic

### PR DESCRIPTION
**Pull request by MarinHadzhiev**
Fix for #1268 

Change Log:
1. Search for TableType in specific schema, not in the whole DB
2. Search for artefact with type TableType not for any artefact with specified name
3. Minor fixes in drop TableType logic
4. Remove cyclic dependancy

### CODE REVIEW

**Comment by MarinHadzhiev** [comment]
depends on https://github.com/eclipse/dirigible/pull/1597

File: modules/engines/engine-hdb/src/main/java/com/sap/xsk/hdb/ds/processors/hdbstructure/HDBSynonymRemover.java [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+
+import com.sap.xsk.hdb.ds.service.manager.IXSKDataStructureManager;
+import com.sap.xsk.hdb.ds.service.manager.XSKSynonymManagerService;
+import com.sap.xsk.utils.XSKHDBUtils;
+import java.sql.Connection;
+import java.sql.SQLException;
+
+public class HDBSynonymRemover {
+
+  IXSKDataStructureManager synonymManagerService;
```
Comment (ThuF) [inline modules/engines/engine-hdb/src/main/java/com/sap/xsk/hdb/ds/processors/hdbstructure/HDBSynonymRemover.java:22]: Why not private?

File: modules/engines/engine-hdb/src/main/java/com/sap/xsk/hdb/ds/processors/hdbstructure/XSKTableTypeDropProcessor.java [hunk 2]
```diff
@@ -11,63 +11,63 @@
...
+  }
+
+  String getDropTableTypeSQL(Connection connection, String tableTypeName) throws IllegalStateException {
+    return SqlFactory.getNative(connection).drop().tableType(tableTypeName).build();
+  }
+
+  boolean tableTypeDoesNotExist(Connection connection, XSKDataStructureHDBTableTypeModel tableTypeModel) throws SQLException {
+    return !SqlFactory.getNative(connection)
+        .exists(connection, tableTypeModel.getSchema(), tableTypeModel.getName(), DatabaseArtifactTypes.TABLE_TYPE);
+  }
```
Comment (ThuF) [inline modules/engines/engine-hdb/src/main/java/com/sap/xsk/hdb/ds/processors/hdbstructure/XSKTableTypeDropProcessor.java:72]: Can these be `private`?

File: modules/engines/engine-hdb/src/main/java/com/sap/xsk/hdb/ds/processors/hdbstructure/HDBSynonymRemover.java (hunk 1 above)
Comment (MarinHadzhiev) [reply modules/engines/engine-hdb/src/main/java/com/sap/xsk/hdb/ds/processors/hdbstructure/HDBSynonymRemover.java]: My bad :).

**Review by ThuF — APPROVED** [review]

**Pull request merged**